migration_guide: add entries for SLM (#12714)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/migration/migration_guide_2.5.rst

* All references to GNSS assistance have been changed from ``A-GPS`` to `A-GNSS`_.
  This affects the :ref:`nrfxlib:nrf_modem_gnss_api`, as well as :ref:`lib_nrf_cloud`, :ref:`lib_location`, and :ref:`lib_lwm2m_location_assistance` libraries.
  See release notes for changes related to each component.

> * For the Serial LTE Modem (SLM) application:
>   * The :kconfig:option:`CONFIG_SLM_AGPS` Kconfig option has been replaced by :kconfig:option:`CONFIG_NRF_CLOUD_AGNSS` and is now enabled by default.

Reviewer: ```suggestion
* For the Serial LTE Modem (SLM) application:

  * The :kconfig:option:`CONFIG_SLM_AGPS` Kconfig option has been replaced by :kconfig:option:`CONFIG_NRF_CLOUD_AGNSS` and is now enabled by default.
```
Breaks doc build by trying to apply term definition formatting.

  * The :kconfig:option:`CONFIG_SLM_PGPS` Kconfig option has been replaced by :kconfig:option:`CONFIG_NRF_CLOUD_PGPS`.
  * The :kconfig:option:`CONFIG_SLM_LOCATION` Kconfig option has been replaced by :kconfig:option:`CONFIG_NRF_CLOUD_LOCATION` and is now enabled by default.
  * The ``#XCELLPOS`` and ``#XWIFIPOS`` AT commands have been replaced by ``#XNRFCLOUDPOS``.

>   * The ``#XAGPS`` and ``#XPGPS``AT commands have been replaced by ``#XGPS``.

Reviewer: ```suggestion
  * The ``#XAGPS`` and ``#XPGPS`` AT commands have been replaced by ``#XGPS``.
```

* Applications that use Zephyr's LwM2M stack and the :ref:`lib_lwm2m_client_utils` library must refactor to use the new event :c:member:`LWM2M_FOTA_UPDATE_MODEM_RECONNECT_REQ` when updating the modem firmware to avoid rebooting the device.
  For an example, see the :ref:`lwm2m_client` sample.
* Applications that use Zephyr's LwM2M stack are recommended to use the :kconfig:option:`CONFIG_LWM2M_UPDATE_PERIOD` Kconfig option to set the LwM2M update sending interval.
* For the Serial LTE Modem (SLM) application:

>   * If you are using the LwM2M carrier library, make sure to take into account the addition of the auto-connect feature that is enabled by default.

Reviewer 2: ```suggestion
  * If you are using the :ref:`liblwm2m_carrier_readme` library, make sure to take into account the addition of the auto-connect feature that is enabled by default.
```
